### ISSUE TITLE
error: segmentio/clear-scripts@*: cannot resolve

### ISSUE THREAD

**Issue by hbbtstar** [segmentio/analytics.js-integrations#641]
Hi! I forked the repository and set up the dev environment as per the readme, but I'm getting an error when I try to run make from inside the directory:

building : test/index.js
        error : Error: segmentio/clear-scripts@*: cannot resolve
    at error (/home/alourenco/WebstormProjects/analytics.js-integrations/node_modules/duo/node_modules/duo-package/node_modules/gh-resolve/index.js:184:10)
    at resolve (/home/alourenco/WebstormProjects/analytics.js-integrations/node_modules/duo/node_modules/duo-package/node_modules/gh-resolve/index.js:66:15)
    at retry (/home/alourenco/WebstormProjects/analytics.js-integrations/node_modules/duo/node_modules/duo-package/node_modules/gh-resolve/index.js:84:7)
    at /home/alourenco/WebstormProjects/analytics.js-integrations/node_modules/duo/node_modules/duo-package/node_modules/gh-resolve/index.js:73:31
    at ChildProcess.exithandler (child_process.js:758:5)
    at ChildProcess.emit (events.js:110:17)
    at maybeClose (child_process.js:1015:16)
    at Socket.<anonymous> (child_process.js:1183:11)
    at Socket.emit (events.js:107:17)
    at Pipe.close (net.js:485:12)

I can't figure out why it's failing to resolve, and google's not being particularly helpful either. Anyone have any suggestions? Thanks!

**Comment by ndhoule** [comment]
Hey @hbbtstar—neither I nor a colleague have been been able to reproduce this on a few different machines. My thought is that it's probably a network issue (maybe a temporary one); is this still an issue for you?

If it is still occurring, do me a favor and try these suggestions/answer these questions and report back:

- What OS are you using?
- Can you do a `make distclean` and then try `make` again?
- If you have access to another network, can you reproduce this on both? (My guess is that this is it.)
- If you have access to another computer, can you reproduce this on it?
- If you have a `~/.netrc`, try moving it and `make distclean ; make` again. Does that fix it?

Just some suggestions off the top of my head; if these don't fix the issue, let me know and we can dig deeper.

**Comment by hbbtstar** [comment]
Hi!

Thanks for this. Sorry for the late reply. 

I'm using Linux Mint 17.1 as my OS. 

the make distclean; make and removing the netrc didn't help, but I was able to solve the problem in a clean VM of Ubuntu (14.04 LTS)- the .netrc never worked for me though in the VM either. I ended up putting the github auth token in .bashrc and that resolved it (on the VM, the host still can't make successfully), if that helps anyone else who runs into a similar issue.

Also I checked the versions of NPM, n, git, curl, etc. from 17.1 against the Ubuntu VM, and they're all the same versions, so I have no idea what the difference might be. 

After doing that, however, I am getting a different error when I run make test (still on the VM): I have not tried to add any custom integrations yet, this is the source as pulled from git.

```
Error loading resource http://localhost:56678/saucelabs.js (203). Details: Error downloading http://localhost:56678/saucelabs.js - server replied: Not Found
Failed to start mocha: Init timeout
Unsafe JavaScript attempt to access frame with URL about:blank from frame with URL file:///home/alourenco/analytics.js-integrations/node_modules/mocha-phantomjs/lib/mocha-phantomjs.coffee. Domains, protocols and ports must match.
```

make test-browser just hangs at a white blank HTML page with this source:

http://pastebin.com/U7FMtQF2

I can open another ticket for this if that's easier to check for you guys! Thanks for the quick turnaround, I know I'm a bit far forward timezone wise :)

**Comment by ndhoule** [comment]
@hbbtstar: Looks like there's an uncaught error before Mocha finishes loading—PhantomJS gives fairly poor feedback here, but the browser console is much more helpful. Usually it's either a syntax error or missing dependencies; since you're tracking master, it's probably missing dependencies. (My guess is, one of the dependencies you errored out on earlier is either missing or incomplete.)

Try running `make distclean` and `make test-browser` again; if that doesn't fix the issue, can you paste a screenshot of your browser console when running `make test-browser`?

**Comment by hbbtstar** [comment]
Hi ndhoule!

that did work, though I had to do it a number of times before it did. I did finally get it working by forcing all of the connections in git to go through SSH instead of HTTPS, and I noticed a couple of strange errors once I did that which led me to the root of the problem- you were right in your initial assessment, it was a network problem (specifically, my ISP's DNS servers). I switched to Google's DNS servers and everything on every machine is completely functional now.

Thank you so much for your help! You guys were fantastic, and I'm really glad the issue has been resolved.
